src/pages/index.js

```diff
@@ -85,30 +96,47 @@ profilePostButton.addEventListener("click", (evt) => {
   disableButton(modalButtonPost, settings);
 });
 
-// Adding event listener to the modal edit save button .
-modalFormEdit.addEventListener("submit", (evt) => {
+// Adding event listener to the modal-post save button .
+modalFormPost.addEventListener("submit", (evt) => {
+  const button = evt.submitter;
+  setButtonText(button, true, "Saving...", "Save");
   evt.preventDefault();
-  profileTitle.textContent = modalEditNameInput.value;
-  profileDescription.textContent = modalEditDescriptionInput.value;
-  closeModal(modalEdit);
-
-  disableButton(modalButtonEdit, settings);
+  api
+    .postCards({ link: modalPostCardImg.value, name: modalPostCardTitle.value })
+    .then((data) => {
+      console.log(data);
+      renderCard(data);
+    })
+    .catch((err) => console.error(err))
+    .finally(() => {
+      closeModal(modalPost);
+      evt.target.reset();
```

Comment:
```js

      closeModal(modalPost);
      evt.target.reset();
```


These 2 lines should be in `.then` in line 107

* You need to close modals only in `then` blocks after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server), and if you change the button text, you could see the change as well. **Please, correct it everywhere in the code**
* You need to **reset** the form only in a `then` block in order not to do that immediately after `submit` because when there is an error, the modal will still be opened, and the user can try to send the data again to the server without having to type the data into the inputs once more.